compaction/compaction_manager.hh

```diff
@@ -380,6 +380,31 @@ public:
     // parameter job is a function that will carry the operation
     future<> run_custom_job(replica::table* t, sstables::compaction_type type, const char *desc, noncopyable_function<future<>(sstables::compaction_data&)> job);
 
+    class compaction_reenabler {
```

Comment:
I'd have named it as something like compaction_disabler, as reenabler is something it does on dtor, whereas disabler is what it does in its ctor, so perhaps compaction_disabled(r)_guard

Reply:
compaction_disabled_guard sounds not too bad...
The Pr is already merged though